pkg/virtctl/guestfs/guestfs.go

```diff
@@ -0,0 +1,483 @@
+func SetClient(f ClientCreator) {
+	createClientFunc = f
+}
+
+// SetDefaulClient sets the default function to create the Kubernetes client
+func SetDefaulClient() {
+	createClientFunc = createClient
+}
+
+// AttacherCreator is a function that attach a command to a pod using the Kubernetes client
+type AttacherCreator func(client *K8sClient, p *corev1.Pod, command string) error
+
+var createAttacherFunc AttacherCreator
+
+// SetAttacher allows overriding the default attacher function. Useful for creating a mock function for the testing.
+func SetAttacher(f AttacherCreator) {
+	createAttacherFunc = f
+}
+
+// SetDefaulAttacher sets the default function to attach to a pod
+func SetDefaulAttacher() {
+	createAttacherFunc = createAttacher
+}
+
+// ImageSet is a function to set the setImage
+type ImageSet func(virtClient kubecli.KubevirtClient) error
+
+var imageSetFunc ImageSet
+
+// SetClient allows overriding the default Kubernetes client. Useful for creating a mock function for the testing.
+func SetImageSetFunc(f ImageSet) {
+	imageSetFunc = f
+}
+
+// SetDefaultImageSet sets the default function to set the image
+func SetDefaultImageSet() {
+	imageSetFunc = setImage
+}
+
+func init() {
+	SetDefaulClient()
+	SetDefaulAttacher()
+	SetDefaultImageSet()
+}
+
+func (c *guestfsCommand) run(cmd *cobra.Command, args []string) error {
+	if pullPolicy != string(corev1.PullAlways) &&
+		pullPolicy != string(corev1.PullNever) &&
+		pullPolicy != string(corev1.PullIfNotPresent) {
+		return fmt.Errorf("Not valide pull policy: %s", pullPolicy)
+	}
+	var inUse bool
+	conf, err := c.clientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	client, err := createClientFunc(conf, c.clientConfig)
+	if err != nil {
+		return err
+	}
+	if err = imageSetFunc(client.VirtClient); err != nil {
+		return err
+	}
+	fmt.Printf("Use image: %s \n", image)
+	exist, _ := client.existsPVC(pvc, namespace)
+	if !exist {
+		return fmt.Errorf("The PVC %s doesn't exist", pvc)
+	}
+	inUse, err = client.isPVCinUse(pvc, namespace)
+	if err != nil {
+		return err
+	}
+	if inUse {
+		return fmt.Errorf("PVC %s is used by another pod", pvc)
+	}
+	isBlock, err := client.isPVCVolumeBlock(pvc, namespace)
+	if err != nil {
+		return err
+	}
+	defer client.removePod(namespace)
+	return client.createInteractivePodWithPVC(pvc, image, namespace, "/entrypoint.sh", []string{}, isBlock)
+}
+
+// K8sClient holds the information of the Kubernetes client
+type K8sClient struct {
+	Client     kubernetes.Interface
+	config     *rest.Config
+	VirtClient kubecli.KubevirtClient
+}
+
+// setImage sets the image name based on the information retrieved by the KubeVirt server. If none of the fields is set, the command will try to use the image libguestfs-tools:latest.
+func setImage(virtClient kubecli.KubevirtClient) error {
+	var imageName string
+	info, err := virtClient.GuestfsVersion().Get()
+	if err != nil {
+		return err
+	}
```

Comment:
do we have any error checking that validates we're given container image information from this endpoint? For example, what if digest, tag, and registry are all empty?

Reply:
I lost the check in the various rebase and change, thanks!

Reply:
I added an error if either the tag or the digest are empty. For the registry, I simply don't set if it is empty, Maybe they want to use unqualified images